### ISSUE TITLE
Fix some security flaws to avoid cheating

### ISSUE THREAD

**Issue by apojomovsky** [apojomovsky/cuatro_en_linea#31]
- When returning values to a public message, return a copy of the object and not the object itself
- You should do that for Gameboard, Player, and Match

**Comment by apojomovsky** [comment]
FYI @basicNew

**Comment by basicNew** [comment]
qq: why Gameboard, Player, and Match ? Shouldn't we only make a copy when playing on a board? i.e. in:

```
    def play_next_turn(self):
        if self._board.is_game_over():
            raise GameIsOver(self.who_won)
        else:
            self._active_player.play(self._board) # <--- HERE copy(self._board)
```

**Comment by apojomovsky** [comment]
Fixed in #46
